common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -165,7 +165,7 @@ public static void submitSilent(@SuppressWarnings(WarningType.rawtype_warning) @
                 public void run() {
                     try {
                         final CommandParameters commandParameters = command.getParameters();
-                        final String correlationId = initializeDiagnosticContext(commandParameters.getCorrelationId(), commandParameters.getSdkType().getProductName(), commandParameters.getSdkVersion());
+                        final String correlationId = initializeDiagnosticContext(commandParameters.getCorrelationId(), commandParameters.getSdkType() == null ? "" : commandParameters.getSdkType().getProductName(), commandParameters.getSdkVersion());
```

Comment:
Are there any other cases where this could happen (besides the test)? and should we pick a better default value for such scenarios?

Reply:
Realistically, we should accept any String that a client sends us and choose to sanitize it accordingly.  If that means accepting anything of a given pattern and converting everything else to UNKNOWN and pushing the transmitted value over telemetry as unknownClientSpecifier = <blah>, that's perfect.

Reply:
No, there wouldn't be any such scenarios, since these values are coded when the requests are being spawned. These exist only in the test case. and with this implementation the test cases are also taken care of and are executed successfully.

Reply:
There is also an additional null check for this field on the broker side, when rebuilding the CommandParameters from teh BrokerRequest.

Reply:
I've made a bug to remove type-sharing from this interface, since it's just a problem waiting to happen.

Reply:
@AdamBJohnsonx yes I agree, since this field is not exposed to user input, hence it is safeguarded by that. And like you suggested there is a refactor task needed with certain fields involved in interaction over wire and once things settle down we can club in this change as well

Reply:
@tanmaymanolkar1 - don't be so trusting. You could get anything over that interface and it shouldn't crash you.

Reply:
@tanmaymanolkar1 I think you can create a constant for `unknown` and use that here....I don't personally like the empty string